File: programming/problem_6/problem_6.dart
void main() {
  print(
    averageExcludingMinMax([1, 2, 3, 4, 5]),
  );
}

double averageExcludingMinMax(List<int> inputNumbers) {
  final min = inputNumbers.reduce((curr, next) => curr < next ? curr : next);
  final max = inputNumbers.reduce((curr, next) => curr > next ? curr : next);
  inputNumbers
    ..removeWhere((e) => e == min)
    ..removeWhere((e) => e == max);
  return inputNumbers.reduce((a, b) => a + b) / inputNumbers.length;
}

File: programming/problem_6/problem_6.rb
def average_excluding_min_max(input_numbers)
    min = input_numbers.min
    max = input_numbers.max
    input_numbers.delete(min)
    input_numbers.delete(max)
    sum = input_numbers.reduce(0) { |prev, curr| prev + curr }
    sum / input_numbers.length.to_f
end

puts 'average_excluding_min_max:', average_excluding_min_max([1, 2, 3, 4, 5])
